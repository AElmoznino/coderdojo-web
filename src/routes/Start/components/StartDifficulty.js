// @flow

import React from 'react'
import ReactMarkdown from 'react-markdown'
import styled from 'styled-components'
import ButtonLink from 'components/Button/ButtonLink'
import H1 from 'components/Typography/H1'
import GridSection from 'components/Grid/GridSection'

type StartDifficultyProps = {
  difficultyDescription: string,
  difficultyId: string,
  difficultyName: string,
}

const InnerWrap = styled.div`
  display: grid;
  grid-template-columns: 200px 1fr;
  grid-column-gap: ${({ theme }) => theme.sizes.medium};
`

const RightWrap = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`

const AlignedButton = styled(ButtonLink)`
  align-self: flex-end;
`

const StartDifficulty = ({
  difficultyDescription,
  difficultyId,
  difficultyName,
}: StartDifficultyProps) => (
  <GridSection>
    <H1>{difficultyName}</H1>
    <InnerWrap>
      <img alt="placeholder alt" src="https://placeimg.com/200/200/tech" />
      <RightWrap>
        <ReactMarkdown>{difficultyDescription}</ReactMarkdown>
        <AlignedButton to={`/overview/${difficultyId}`}>
          Kom ig√•ng
        </AlignedButton>
      </RightWrap>
    </InnerWrap>
  </GridSection>
)

export default StartDifficulty
