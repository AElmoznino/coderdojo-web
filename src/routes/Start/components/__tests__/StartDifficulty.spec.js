import React from 'react'
import { shallow } from 'enzyme'
import StartDifficulty from '../StartDifficulty'

describe('components/StartDifficulty', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <StartDifficulty
        difficultyDescription="testText"
        difficultyId="testLink"
        difficultyName="testTitle"
      />,
    )
  })

  it('renders StartDifficulty', () => {
    expect(component).toMatchSnapshot()
  })
})
