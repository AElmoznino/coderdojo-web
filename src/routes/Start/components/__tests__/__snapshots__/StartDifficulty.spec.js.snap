// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StartDifficulty renders StartDifficulty 1`] = `
<styled.div>
  <styled.h1>
    testTitle
  </styled.h1>
  <styled.div>
    <img
      alt="placeholder alt"
      src="https://placeimg.com/200/200/tech"
    />
    <styled.div>
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      >
        testText
      </ReactMarkdown>
      <Styled(Styled(Link))
        to="/overview/testLink"
      >
        Kom ig√•ng
      </Styled(Styled(Link))>
    </styled.div>
  </styled.div>
</styled.div>
`;
